src/VisualStudio/VisualBasic/Impl/ProjectSystemShim/EntryPointFinder.vb

```diff
@@ -26,7 +26,11 @@ Namespace Microsoft.VisualStudio.LanguageServices.VisualBasic.ProjectSystemShim
         Public Shared Function FindEntryPoints(symbol As INamespaceSymbol, findFormsOnly As Boolean) As IEnumerable(Of INamedTypeSymbol)
             Dim visitor = New EntryPointFinder(findFormsOnly)
             ' Only search source symbols
-            visitor.Visit(symbol.ContainingCompilation.SourceModule.GlobalNamespace)
+            If symbol.ContainingCompilation IsNot Nothing Then
+                symbol = symbol.ContainingCompilation.SourceModule.GlobalNamespace
```

Comment:
why is the VB side not the same?

Reply:
This doesn't seem to be resolved -- VB we're still doing this odd "hop back to compilation's namespace" when we aren't in C#

Reply:
Sorry, I resolved this by mistake.

This is because in C# we never consider Forms to be entry points like we do in VB.

In CPS If we don't do this filtering, we get the following:
<img width="390" alt="image" src="https://user-images.githubusercontent.com/9797472/162845069-15f5d0c8-b567-44e2-be21-a73cc0839f8b.png">

By filtering to the source module, we get this:

<img width="233" alt="image" src="https://user-images.githubusercontent.com/9797472/162844651-6a03583a-f2cb-4015-a1b1-f21779ad16b2.png">

We could instead keep the code the same and change the logic on the project system side: https://github.com/dotnet/project-system/blob/fa538031edbbc707fe4c98f79b6e6f5e52e52086/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/StartupObjectsEnumProvider.cs#L82 to pass us a namespace off of the `SourceModule`.

However, if we are going to do that I would prefer we just change the API for [`IEntryPointFinderService`](https://github.com/dotnet/roslyn/blob/main/src/VisualStudio/Core/Def/ProjectSystem/IEntryPointFinderService.cs) to accept a compilation instead of an INamespaceSymbol and avoid this class of error altogether.

Reply:
Ah, I'm getting the background now -- we need that in the VB case because we're being passed in the wrong symbol for the project system side of things. And in the C# case the code is also wrong, but it just doesn't matter since the ref assemblies are unlikely to have a random Main() method that matches.

Should a comment exist on the C# side why there isn't the equivalent pattern? Or, instead of doing this, just fix the project system to pass in the source global namespace too?

Reply:
Happy to add a comment. I really would prefer we just did this https://github.com/dotnet/roslyn/pull/60614 and changed up the Proejct System code to pass use a compilation instead. However I can be reasoned with.

Reply:
Well, this conversation has gone longer than fixing the project system so.... :smile:. But I'm OK either way.

Reply:
yeah, this is what I get for trying to make a quick PR...

Reply:
alright added a comment explaining the divergence for now. filed issue https://github.com/dotnet/roslyn/issues/60696 to track updating the API